### PR TITLE
Nextjs 13 setup - Skill tree

### PR DESCRIPTION
**Pull request by arpit01923**
### Description:
Setup project with eslint, prettier and pre-commit hook

### Anything you would like to inform the reviewer about:
Added
- Eslint
- Prettier 
- Pre-commit hooks

### Dev Tested:
- None

### Test Stats
- None

### Images/video of the change:
image

### Follow-up Issues (if any)
Not right now

### ETA
18/7/2023

### CODE REVIEW

File: src/app/page.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
+import Image from "next/image";
+
+export default function Home() {
+  return (
+    <main className="flex min-h-screen flex-col items-center justify-between p-24">
+      <div className="z-10 w-full max-w-5xl items-center justify-between font-mono text-sm lg:flex">
+        <p className="fixed left-0 top-0 flex w-full justify-center border-b border-gray-300 bg-gradient-to-b from-zinc-200 pb-6 pt-8 backdrop-blur-2xl dark:border-neutral-800 dark:bg-zinc-800/30 dark:from-inherit lg:static lg:w-auto  lg:rounded-xl lg:border lg:bg-gray-200 lg:p-4 lg:dark:bg-zinc-800/30">
```
Comment (bhtibrewal) [inline src/app/page.tsx:7]: please remove the file, code and images that is will not be need

**Review by bhtibrewal — CHANGES_REQUESTED** [review]

**Review by bhtibrewal — COMMENTED** [review]
I cannot find the pre-commit hook, and also add a PR template to the project

File: src/app/page.tsx (hunk 1 above)
Comment (arpit01923) [reply src/app/page.tsx]: Removed unnecessary, image and code

**Review by bhtibrewal — APPROVED** [review]
LGTM

File: .vscode/settings.json
Comment (RitikJaiswal75) [inline .vscode/settings.json:1]: I think .vscode should be in gitignore, we do not want modifying developers' vs code setting

File: package.json
Comment (RitikJaiswal75) [inline package.json:1]: Please segregate dependencies and dev dependencies properly

File: pull_request_template.md
Comment (RitikJaiswal75) [inline pull_request_template.md:1]: Do we want all of our pull requests to this repo to look like this?
If no, do we need this file?

Description here is too specific to this pr its not generic to be made a template and enforce it over all developers.

File: src/app/globals.css
Comment (RitikJaiswal75) [inline src/app/globals.css:1]: lets set the padding and margin to 0 here for root.

File: tsconfig.json
Comment (RitikJaiswal75) [inline tsconfig.json:1]: can we add an outdir here?
or is it managed by next js automatically?

**Review by RitikJaiswal75 — CHANGES_REQUESTED** [review]

File: .vscode/settings.json
Comment (arpit01923) [reply .vscode/settings.json:1]: Added .vscode in gitignore

File: package.json
Comment (arpit01923) [reply package.json:1]: Segregated dependencies and dev dependencies

File: src/app/globals.css
Comment (arpit01923) [reply src/app/globals.css:1]: Added padding and margin to 0

File: tsconfig.json
Comment (arpit01923) [reply tsconfig.json:1]: In a Next.js project, the output directory (outdir) is managed automatically.

File: pull_request_template.md
Comment (arpit01923) [reply pull_request_template.md:1]: A standardized template for pull requests can provide clarity and consistency in the information shared during the review process.
If the template doesn't add value to your team's workflow, I will  removing the **pull_request_template.md**

Comment (RitikJaiswal75) [reply pull_request_template.md:1]: I agree with having a template, but it should be generic. The one here is very specific to your PR.

File: .vscode/settings.json
Comment (RitikJaiswal75) [reply .vscode/settings.json:1]: but the file is still present, remove it from the pull

Comment (arpit01923) [reply .vscode/settings.json:1]: Removed **.vscode** @RitikJaiswal75

File: pull_request_template.md
Comment (arpit01923) [reply pull_request_template.md:1]: Ok got it @RitikJaiswal75. Removed the file

Comment (bhtibrewal) [reply pull_request_template.md:1]:
```
### Issue:

### Description:

### Anything you would like to inform the reviewer about:

### Dev Tested:
- [ ] Yes

### Test Stats

### Images/video of the change:

### Follow-up Issues (if any)
```

please add this PR template

**Review by RitikJaiswal75 — APPROVED** [review]

**Review by vvaibhavdesai — APPROVED** [review]

**Review by bhtibrewal — APPROVED** [review]

**Pull request merged**